source/text/Format.cpp

```diff
@@ -43,6 +43,36 @@ namespace {
 
 		reverse(result.begin(), result.end());
 	}
+
+	// Helper function for ExpandConditions.
+	//
+	// source.substr(formatStart, formatSize) contains the format (credits, mass, etc)
+	// source.substr(conditionStart, conditionSize) contains the condition name
+	//
+	// If formatStart or formatSize are string::npos, then there is no formatting.
+	//
+	// The condition value, obtained via getter(), is sent through Format::Whatever()
+	// and the output of that is appended to the result.
+	void AppendCondition(string &result, const string &source, Format::ConditionGetter getter,
+		size_t formatStart, size_t formatSize, size_t conditionStart, size_t conditionSize)
+	{
+		int64_t value = getter(source, conditionStart, conditionSize);
+		if(formatStart == string::npos || formatSize == string::npos)
+			result.append(Format::Number(value));
+		else if(!source.compare(formatStart, formatSize, "raw"))
```

Comment:
Can you extract the substring to its own variable and then compare that with `==`?
I feel like that would be much easier to read.